Duplicated `properties` tag in the `pom` breaks publishing

See https://github.com/scala/scala3/actions/runs/8430164281

```
2024-03-26 05:12:26.505+0100 error [SonatypeClient] [close] Failed  - (SonatypeClient.scala:181)
2024-03-26 05:12:26.506+0100 error [SonatypeClient] Activity name:close, started:2024-03-26T04:11:46.157Z, stopped:2024-03-26T04:12:22.966Z  - (SonatypeClient.scala:469)
2024-03-26 05:12:26.508+0100 error [SonatypeClient]     Failed: sources-staging, failureMessage:Illegal POM: /org/scala-lang/scala3-library_sjs1_3/3.4.2-RC1-bin-20240325-198750b-NIGHTLY/scala3-library_sjs1_3-3.4.2-RC1-bin-20240325-198750b-NIGHTLY.pom Duplicated tag: 'properties' (position: START_TAG seen ...&lt;/properties>\n    &lt;properties>... @96:17)   - (SonatypeClient.scala:387)
2024-03-26 05:12:26.508+0100 error [SonatypeClient]     Failed: javadoc-staging, failureMessage:Illegal POM: /org/scala-lang/scala3-library_sjs1_3/3.4.2-RC1-bin-20240325-198750b-NIGHTLY/scala3-library_sjs1_3-3.4.2-RC1-bin-20240325-198750b-NIGHTLY.pom Duplicated tag: 'properties' (position: START_TAG seen ...&lt;/properties>\n    &lt;properties>... @96:17)   - (SonatypeClient.scala:387)
2024-03-26 05:12:26.508+0100 error [SonatypeClient]     Failed: pom-staging, failureMessage:Invalid POM: /org/scala-lang/scala3-library_sjs1_3/3.4.2-RC1-bin-20240325-198750b-NIGHTLY/scala3-library_sjs1_3-3.4.2-RC1-bin-20240325-198750b-NIGHTLY.pom: Parsing Error: Duplicated tag: 'properties' (position: START_TAG seen ...&lt;/properties>\n    &lt;properties>... @96:17)   - (SonatypeClient.scala:387)
2024-03-26 05:12:26.510+0100 error [Sonatype] [STAGE_FAILURE] Failed to close the repository.  - (Sonatype.scala:440)
Error: Process completed with exit code 1.
```

Seems like we have a duplicated properties tag in the `pom.xml`, which breaks publishing...
I see a likely culprit here:
https://github.com/scala/scala3/commit/18b146bede042eacca7e2d77941ffb54ff68b453

@Kordyjan can you have a look?

--- a/project/Build.scala
+++ b/project/Build.scala
@@ -1999,6 +1999,10 @@ object Build {
     publishTo := sonatypePublishToBundle.value,
     publishConfiguration ~= (_.withOverwrite(true)),
     publishLocalConfiguration ~= (_.withOverwrite(true)),
+    projectID ~= {id =>
+      val line = "scala.versionLine" -> versionLine
+      id.withExtraAttributes(id.extraAttributes + line)
+    },
     Test / publishArtifact := false,
     homepage := Some(url(dottyGithubUrl)),
     licenses += (("Apache-2.0",
@@ -2009,10 +2013,6 @@ object Build {
         "scm:git:git@github.com:scala/scala3.git"
       )
     ),
-    pomExtra :=
-      <properties>
-        <scala.versionLine>{versionLine}</scala.versionLine>
-      </properties>,
     developers := List(
       Developer(
         id = "odersky",

